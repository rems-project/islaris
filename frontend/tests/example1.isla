(trace
  (declare-const v3428 (_ BitVec 64))
  (define-const v3429 v3428)
  (read-reg |R30| nil v3429)
  (define-const v3435 v3429)
  (define-const v3436 v3435)
  (branch-address v3436)
  (define-const v3437 v3435)
  (write-reg |_PC| nil v3437)
  (write-reg |__PC_changed| nil true))
