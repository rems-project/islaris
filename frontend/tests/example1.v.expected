From isla Require Import isla_lang.

Definition trace : trc := [
  [
    Smt (DeclareConst 3428%Z (Ty_BitVec 64%N)) Mk_annot;
    Smt (DefineConst 3429%Z (Val (Val_Symbolic 3428%Z) Mk_annot)) Mk_annot;
    ReadReg "R30" [] (Val_Symbolic 3429%Z) Mk_annot;
    Smt (DefineConst 3435%Z (Val (Val_Symbolic 3429%Z) Mk_annot)) Mk_annot;
    Smt (DefineConst 3436%Z (Val (Val_Symbolic 3435%Z) Mk_annot)) Mk_annot;
    BranchAddress (Val_Symbolic 3436%Z) Mk_annot;
    Smt (DefineConst 3437%Z (Val (Val_Symbolic 3435%Z) Mk_annot)) Mk_annot;
    WriteReg "_PC" [] (Val_Symbolic 3437%Z) Mk_annot;
    WriteReg "__PC_changed" [] (Val_Bool true) Mk_annot
  ]
].
